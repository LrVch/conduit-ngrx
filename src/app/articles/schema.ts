import { schema } from 'normalizr';

export const articleSchema = new schema.Entity(
  'articles',
  {},
  { idAttribute: 'slug' }
);
export const responceArticleSchema = { article: articleSchema };
export const responceArticlesSchema = { articles: [articleSchema] };
