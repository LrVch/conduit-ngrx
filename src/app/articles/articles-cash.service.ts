import { Injectable } from '@angular/core';
import * as schema from './schema';
import { normalize } from 'normalizr';
import { Article } from '@app/core';

export interface ResponseArticles {
  articles: Article[];
  articlesCount: number;
}

export interface NormalizedArticlesResponse {
  entities: {
    articles: {
      [key: string]: Article;
    };
  };
  result: {
    articles: string[];
    articlesCount: number;
  };
}

@Injectable({
  providedIn: 'root'
})
export class ArticlesCashService {
  public static normalizeArticlesResponce(
    articles: ResponseArticles
  ): NormalizedArticlesResponse {
    return normalize(articles, schema.responceArticlesSchema);
  }

  constructor() {}
}
