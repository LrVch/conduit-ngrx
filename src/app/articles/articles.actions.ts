import { Action } from '@ngrx/store';
import { Article, Errors } from '@app/core';
import { Articles } from '@app/core/models/articles.model';
import { ArticlesConfigState } from './articlesConfig.reducer';

export enum ArticlesActionTypes {
  /* Articles */
  LoadArticlesRequest = '[Articles] Load Articles Request',
  LoadArticlesLoaded = '[Articles] Load Articles Loaded',
  LoadArticlesCompleted = '[Articles] Load Articles Completed',
  LoadArticlesFail = '[Articles] Load Articles Fail',
  SetReturnArticlesConfig = '[Articles] Set Return Articles Config',
  ClearReturnArticlesConfig = '[Articles] Clear Return Articles Config',

  /* Article */
  ToggleArticleFavoriteRequest = '[Article] Toggle Article Favorite Request',
  ToggleArticleFavoriteSuccess = '[Article] Toggle Article Favorite Success',
  ToggleArticleFavoriteFail = '[Article] Toggle Article Favorite Fail',
  UnFavoriteArticleRequest = '[Article] Un Favorite Article Request',
  FavoriteArticleRequest = '[Article] Favorite Article Request',
  SetFavoritingArticle = '[Article] Set Favoriting Article',
  ClearFavoritingArticle = '[Article] Clear Favoriting Article',

  ArticlesDeleteArticleRequest = '[Articles] Delete Article Request',
  ArticlesDeleteArticleSuccess = '[Articles] Delete Article Success',
  ArticlesDeleteArticleFail = '[Articles] Delete Article Fail',
  ArticlesDeleteArticleConfirmationRequest = '[Articles] Article Delete Confirmation Request',
  ArticlesDeleteArticleConfirmation = '[Articles] Article Delete Confirmation',

  /* Tags */
  LoadTagsRequest = '[Articles] Load Tags Request',
  LoadTagsLoaded = '[Articles] Load Tags Loaded',
  LoadTagsCompleted = '[Articles] Load Tags Completed',
  LoadTagsFail = '[Articles] Load Tags Fail',

  /* Articles config */
  SetTag = '[Articles] Set Tag',
  SetAuthor = '[Articles] Set Author',
  SetOffset = '[Articles] Set Offset',
  SetLimit = '[Articles] Set Limit',
  SetPageIndex = '[Articles] Set Page Index',
  SetFavorited = '[Articles] Set Favorited',
  SetCurrentPage = '[Articles] Set Current Page',
  SetTypeOfFeed = '[Articles] Set Type Of Feed',
  ResetConfig = '[Articles] Reset Config',
}

/* Articles */
export class ClearReturnArticlesConfig implements Action {
  readonly type = ArticlesActionTypes.ClearReturnArticlesConfig;
}

export class SetReturnArticlesConfig implements Action {
  readonly type = ArticlesActionTypes.SetReturnArticlesConfig;

  constructor(public payload: { config: ArticlesConfigState }) { }
}

export class LoadArticlesRequest implements Action {
  readonly type = ArticlesActionTypes.LoadArticlesRequest;
}

export class LoadArticlesLoaded implements Action {
  readonly type = ArticlesActionTypes.LoadArticlesLoaded;

  constructor(public payload: { articles: Articles }) { }
}

export class LoadArticlesCompleted implements Action {
  readonly type = ArticlesActionTypes.LoadArticlesCompleted;
}

export class LoadArticlesFail implements Action {
  readonly type = ArticlesActionTypes.LoadArticlesFail;
}

export class SetCurrentPage implements Action {
  readonly type = ArticlesActionTypes.SetCurrentPage;

  constructor(public payload: { currentPage: number }) { }
}

/* Article */
export class ToggleArticleFavoriteRequest implements Action {
  readonly type = ArticlesActionTypes.ToggleArticleFavoriteRequest;

  constructor(public payload: { article: Article }) { }
}

export class ToggleArticleFavoriteSuccess implements Action {
  readonly type = ArticlesActionTypes.ToggleArticleFavoriteSuccess;

  constructor(public payload: { article: Article, showNotification?: boolean }) { }
}

export class ToggleArticleFavoriteFail implements Action {
  readonly type = ArticlesActionTypes.ToggleArticleFavoriteFail;

  constructor(public payload: { article: Article, showNotification?: boolean  }) { }
}

export class UnFavoriteArticleRequest implements Action {
  readonly type = ArticlesActionTypes.UnFavoriteArticleRequest;

  constructor(public payload: { article: Article }) { }
}

export class FavoriteArticleRequest implements Action {
  readonly type = ArticlesActionTypes.FavoriteArticleRequest;

  constructor(public payload: { article: Article }) { }
}

export class SetFavoritingArticle implements Action {
  readonly type = ArticlesActionTypes.SetFavoritingArticle;

  constructor(public payload: { article: Article }) { }
}

export class ClearFavoritingArticle implements Action {
  readonly type = ArticlesActionTypes.ClearFavoritingArticle;
}

export class ArticlesDeleteArticleRequest implements Action {
  readonly type = ArticlesActionTypes.ArticlesDeleteArticleRequest;

  constructor(public payload: { article: Article }) { }
}

export class ArticlesDeleteArticleSuccess implements Action {
  readonly type = ArticlesActionTypes.ArticlesDeleteArticleSuccess;

  constructor(public payload?: { article?: Article }) { }
}

export class ArticlesDeleteArticleFail implements Action {
  readonly type = ArticlesActionTypes.ArticlesDeleteArticleFail;

  constructor(public payload: { article: Article }) { }
}

export class ArticlesDeleteArticleConfirmationRequest implements Action {
  readonly type = ArticlesActionTypes.ArticlesDeleteArticleConfirmationRequest;

  constructor(public payload: { article: Article, question: string }) { }
}

export class ArticlesDeleteArticleConfirmation implements Action {
  readonly type = ArticlesActionTypes.ArticlesDeleteArticleConfirmation;

  constructor(public payload: { article: Article}) { }
}




/* Tags */
export class LoadTagsRequest implements Action {
  readonly type = ArticlesActionTypes.LoadTagsRequest;
}

export class LoadTagsLoaded implements Action {
  readonly type = ArticlesActionTypes.LoadTagsLoaded;

  constructor(public payload: { tags: string[] }) { }
}

export class LoadTagsCompleted implements Action {
  readonly type = ArticlesActionTypes.LoadTagsCompleted;
}

export class LoadTagsFail implements Action {
  readonly type = ArticlesActionTypes.LoadTagsFail;
}


/* Config */
export class SetTag implements Action {
  readonly type = ArticlesActionTypes.SetTag;

  constructor(public payload: { tag: string }) { }
}

export class SetTypeOfFeed implements Action {
  readonly type = ArticlesActionTypes.SetTypeOfFeed;

  constructor(public payload: { type: string }) { }
}

export class SetAuthor implements Action {
  readonly type = ArticlesActionTypes.SetAuthor;

  constructor(public payload: { author: string }) { }
}

export class SetOffset implements Action {
  readonly type = ArticlesActionTypes.SetOffset;

  constructor(public payload: { offset: number }) { }
}

export class SetLimit implements Action {
  readonly type = ArticlesActionTypes.SetLimit;

  constructor(public payload: { limit: number }) { }
}

export class SetPageIndex implements Action {
  readonly type = ArticlesActionTypes.SetPageIndex;

  constructor(public payload: { pageIndex: number }) { }
}

export class SetFavorited implements Action {
  readonly type = ArticlesActionTypes.SetFavorited;

  constructor(public payload: { favorited: string }) { }
}

export class ResetConfig implements Action {
  readonly type = ArticlesActionTypes.ResetConfig;
}

export type ArticlesActions =
  /* Tags */
  | LoadTagsRequest
  | LoadTagsLoaded
  | LoadTagsCompleted
  | LoadTagsFail
  | SetCurrentPage
  /* Articles */
  | ClearReturnArticlesConfig
  | SetReturnArticlesConfig
  | LoadArticlesRequest
  | LoadArticlesLoaded
  | LoadArticlesCompleted
  | LoadArticlesFail
  /* Article */
  | ToggleArticleFavoriteRequest
  | ToggleArticleFavoriteSuccess
  | ToggleArticleFavoriteFail
  | UnFavoriteArticleRequest
  | FavoriteArticleRequest
  | SetFavoritingArticle
  | ClearFavoritingArticle
  | ArticlesDeleteArticleRequest
  | ArticlesDeleteArticleSuccess
  | ArticlesDeleteArticleFail
  | ArticlesDeleteArticleConfirmationRequest
  | ArticlesDeleteArticleConfirmation
  /* Articles config */
  | SetTypeOfFeed
  | SetTag
  | SetAuthor
  | SetOffset
  | SetLimit
  | SetPageIndex
  | SetFavorited
  | ResetConfig;
