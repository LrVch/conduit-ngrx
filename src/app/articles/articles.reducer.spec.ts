import * as fromArticles from './articles.reducer';
import * as fromArticlesConfig from './articlesConfig.reducer';
import {
  LoadArticlesRequest,
  LoadArticlesLoaded,
  LoadArticlesFail,
  ToggleArticleFavoriteRequest,
  ToggleArticleFavoriteSuccess,
  ToggleArticleFavoriteFail,
  SetReturnArticlesConfig,
  ClearReturnArticlesConfig,
  SetFavoritingArticle,
  ClearFavoritingArticle,
  LoadTagsRequest,
  LoadTagsLoaded,
  LoadTagsFail
} from './articles.actions';
import { getArticle } from '../lib/testing';

describe('Article Reducer', () => {
  const article = getArticle();
  const initialArticlesState: fromArticles.ArticlesState = {
    ...fromArticles.initialState,
    items: { 'slug': article, '1': article }, ids: ['slug', '1'], articlesCount: 2
  };
  const returnConfig = fromArticlesConfig.initialState;

  describe('State changes', () => {
    it('should have an initial state', () => {
      const state = fromArticles.articlesReducer(initialArticlesState, { type: '@@init' } as any);

      expect(state).toBe(initialArticlesState);
    });

    it('should clear items, ids, set loadingArticles as true on "LoadArticlesRequest"', () => {
      const init = { type: '@@init' } as any;
      const action = new LoadArticlesRequest();

      const state = [init, action].reduce(fromArticles.articlesReducer, initialArticlesState);

      expect(state).toMatchSnapshot();
    });

    it('should set items, ids, articlesCount, set loadingArticles as false on "LoadArticlesLoaded"', () => {
      const init = { type: '@@init' } as any;
      const action = new LoadArticlesLoaded({ articles: { items: { '0': article, '1': article }, ids: ['0', '1'], articlesCount: 2 } });

      const state = [init, action].reduce(fromArticles.articlesReducer, initialArticlesState);

      expect(state).toMatchSnapshot();
    });

    it('should set errorLoadingArticles as true on "LoadArticlesFail"', () => {
      const init = { type: '@@init' } as any;
      const action = new LoadArticlesFail();

      const state = [init, action].reduce(fromArticles.articlesReducer, initialArticlesState);

      expect(state).toMatchSnapshot();
    });

    it('should set article favoriting as true on "ToggleArticleFavoriteRequest"', () => {
      const init = { type: '@@init' } as any;
      const action = new ToggleArticleFavoriteRequest({ article });

      const state = [init, action].reduce(fromArticles.articlesReducer, initialArticlesState);

      expect(state).toMatchSnapshot();
    });

    it('should set article favoriting as false on "ToggleArticleFavoriteSuccess"', () => {
      const init = { type: '@@init' } as any;
      const action = new ToggleArticleFavoriteSuccess({ article });

      const state = [init, action].reduce(fromArticles.articlesReducer, initialArticlesState);

      expect(state).toMatchSnapshot();
    });

    it('should set article favoriting as false on "ToggleArticleFavoriteFail"', () => {
      const init = { type: '@@init' } as any;
      const action = new ToggleArticleFavoriteFail({ article });

      const state = [init, action].reduce(fromArticles.articlesReducer, initialArticlesState);

      expect(state).toMatchSnapshot();
    });

    it('should set returnConfig on "SetReturnArticlesConfig"', () => {
      const init = { type: '@@init' } as any;
      const action = new SetReturnArticlesConfig({ config: returnConfig });

      const state = [init, action].reduce(fromArticles.articlesReducer, initialArticlesState);

      expect(state).toMatchSnapshot();
    });

    it('should remove returnConfig on "ClearReturnArticlesConfig"', () => {
      const init = { type: '@@init' } as any;
      const action = new ClearReturnArticlesConfig();

      const state = [init, action].reduce(fromArticles.articlesReducer, initialArticlesState);

      expect(state).toMatchSnapshot();
    });

    it('should set favoritingArticle on "SetFavoritingArticle"', () => {
      const init = { type: '@@init' } as any;
      const action = new SetFavoritingArticle({ article });

      const state = [init, action].reduce(fromArticles.articlesReducer, initialArticlesState);

      expect(state).toMatchSnapshot();
    });

    it('should remove favoritingArticle on "ClearFavoritingArticle"', () => {
      const init = { type: '@@init' } as any;
      const action = new ClearFavoritingArticle();

      const state = [init, action].reduce(fromArticles.articlesReducer, initialArticlesState);

      expect(state).toMatchSnapshot();
    });

    it('should remove favoritingArticle on "ClearFavoritingArticle"', () => {
      const init = { type: '@@init' } as any;
      const action = new ClearFavoritingArticle();

      const state = [init, action].reduce(fromArticles.articlesReducer, initialArticlesState);

      expect(state).toMatchSnapshot();
    });

    it('should set loadingTags as true on "LoadTagsRequest"', () => {
      const init = { type: '@@init' } as any;
      const action = new LoadTagsRequest();

      const state = [init, action].reduce(fromArticles.articlesReducer, initialArticlesState);

      expect(state).toMatchSnapshot();
    });

    it('should set tags on "LoadTagsLoaded"', () => {
      const init = { type: '@@init' } as any;
      const action = new LoadTagsLoaded({ tags: ['one', 'two'] });

      const state = [init, action].reduce(fromArticles.articlesReducer, initialArticlesState);

      expect(state).toMatchSnapshot();
    });

    it('should set errorLoadingTags as true on "LoadTagsFail"', () => {
      const init = { type: '@@init' } as any;
      const action = new LoadTagsFail();

      const state = [init, action].reduce(fromArticles.articlesReducer, initialArticlesState);

      expect(state).toMatchSnapshot();
    });
  });
});
