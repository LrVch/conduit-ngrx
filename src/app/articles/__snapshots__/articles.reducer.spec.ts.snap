// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Reducer State changes should clear items, ids, set loadingArticles as true on "LoadArticlesRequest" 1`] = `
Object {
  "articlesCount": 2,
  "currentPage": 1,
  "errorLoadingArticles": false,
  "errorLoadingTags": false,
  "favoritingArticle": null,
  "ids": Array [],
  "items": Object {},
  "loadingArticles": true,
  "loadingTags": false,
  "returnConfig": null,
  "tags": Array [],
}
`;

exports[`Article Reducer State changes should remove favoritingArticle on "ClearFavoritingArticle" 1`] = `
Object {
  "articlesCount": 2,
  "currentPage": 1,
  "errorLoadingArticles": false,
  "errorLoadingTags": false,
  "favoritingArticle": null,
  "ids": Array [
    "slug",
    "1",
  ],
  "items": Object {
    "1": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
    "slug": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
  },
  "loadingArticles": false,
  "loadingTags": false,
  "returnConfig": null,
  "tags": Array [],
}
`;

exports[`Article Reducer State changes should remove favoritingArticle on "ClearFavoritingArticle" 2`] = `
Object {
  "articlesCount": 2,
  "currentPage": 1,
  "errorLoadingArticles": false,
  "errorLoadingTags": false,
  "favoritingArticle": null,
  "ids": Array [
    "slug",
    "1",
  ],
  "items": Object {
    "1": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
    "slug": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
  },
  "loadingArticles": false,
  "loadingTags": false,
  "returnConfig": null,
  "tags": Array [],
}
`;

exports[`Article Reducer State changes should remove returnConfig on "ClearReturnArticlesConfig" 1`] = `
Object {
  "articlesCount": 2,
  "currentPage": 1,
  "errorLoadingArticles": false,
  "errorLoadingTags": false,
  "favoritingArticle": null,
  "ids": Array [
    "slug",
    "1",
  ],
  "items": Object {
    "1": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
    "slug": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
  },
  "loadingArticles": false,
  "loadingTags": false,
  "returnConfig": null,
  "tags": Array [],
}
`;

exports[`Article Reducer State changes should set article favoriting as false on "ToggleArticleFavoriteFail" 1`] = `
Object {
  "articlesCount": 2,
  "currentPage": 1,
  "errorLoadingArticles": false,
  "errorLoadingTags": false,
  "favoritingArticle": null,
  "ids": Array [
    "slug",
    "1",
  ],
  "items": Object {
    "1": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
    "slug": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
  },
  "loadingArticles": false,
  "loadingTags": false,
  "returnConfig": null,
  "tags": Array [],
}
`;

exports[`Article Reducer State changes should set article favoriting as false on "ToggleArticleFavoriteSuccess" 1`] = `
Object {
  "articlesCount": 2,
  "currentPage": 1,
  "errorLoadingArticles": false,
  "errorLoadingTags": false,
  "favoritingArticle": null,
  "ids": Array [
    "slug",
    "1",
  ],
  "items": Object {
    "1": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
    "slug": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
  },
  "loadingArticles": false,
  "loadingTags": false,
  "returnConfig": null,
  "tags": Array [],
}
`;

exports[`Article Reducer State changes should set article favoriting as true on "ToggleArticleFavoriteRequest" 1`] = `
Object {
  "articlesCount": 2,
  "currentPage": 1,
  "errorLoadingArticles": false,
  "errorLoadingTags": false,
  "favoritingArticle": null,
  "ids": Array [
    "slug",
    "1",
  ],
  "items": Object {
    "1": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
    "slug": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": true,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
  },
  "loadingArticles": false,
  "loadingTags": false,
  "returnConfig": null,
  "tags": Array [],
}
`;

exports[`Article Reducer State changes should set errorLoadingArticles as true on "LoadArticlesFail" 1`] = `
Object {
  "articlesCount": 2,
  "currentPage": 1,
  "errorLoadingArticles": true,
  "errorLoadingTags": false,
  "favoritingArticle": null,
  "ids": Array [
    "slug",
    "1",
  ],
  "items": Object {
    "1": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
    "slug": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
  },
  "loadingArticles": false,
  "loadingTags": false,
  "returnConfig": null,
  "tags": Array [],
}
`;

exports[`Article Reducer State changes should set errorLoadingTags as true on "LoadTagsFail" 1`] = `
Object {
  "articlesCount": 2,
  "currentPage": 1,
  "errorLoadingArticles": false,
  "errorLoadingTags": true,
  "favoritingArticle": null,
  "ids": Array [
    "slug",
    "1",
  ],
  "items": Object {
    "1": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
    "slug": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
  },
  "loadingArticles": false,
  "loadingTags": false,
  "returnConfig": null,
  "tags": Array [],
}
`;

exports[`Article Reducer State changes should set favoritingArticle on "SetFavoritingArticle" 1`] = `
Object {
  "articlesCount": 2,
  "currentPage": 1,
  "errorLoadingArticles": false,
  "errorLoadingTags": false,
  "favoritingArticle": Object {
    "author": Object {
      "bio": "bio",
      "following": false,
      "image": "image",
      "isFollowLoading": false,
      "username": "username",
    },
    "body": "body",
    "createdAt": "1544505543896",
    "description": "desc",
    "favorited": false,
    "favoritesCount": 0,
    "favoriting": false,
    "slug": "slug",
    "tagList": Array [
      "list",
    ],
    "title": "title",
    "updatedAt": "1544505523896",
  },
  "ids": Array [
    "slug",
    "1",
  ],
  "items": Object {
    "1": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
    "slug": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
  },
  "loadingArticles": false,
  "loadingTags": false,
  "returnConfig": null,
  "tags": Array [],
}
`;

exports[`Article Reducer State changes should set items, ids, articlesCount, set loadingArticles as false on "LoadArticlesLoaded" 1`] = `
Object {
  "articlesCount": 2,
  "currentPage": 1,
  "errorLoadingArticles": false,
  "errorLoadingTags": false,
  "favoritingArticle": null,
  "ids": Array [
    "0",
    "1",
  ],
  "items": Object {
    "0": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
    "1": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
  },
  "loadingArticles": false,
  "loadingTags": false,
  "returnConfig": null,
  "tags": Array [],
}
`;

exports[`Article Reducer State changes should set loadingTags as true on "LoadTagsRequest" 1`] = `
Object {
  "articlesCount": 2,
  "currentPage": 1,
  "errorLoadingArticles": false,
  "errorLoadingTags": false,
  "favoritingArticle": null,
  "ids": Array [
    "slug",
    "1",
  ],
  "items": Object {
    "1": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
    "slug": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
  },
  "loadingArticles": false,
  "loadingTags": true,
  "returnConfig": null,
  "tags": Array [],
}
`;

exports[`Article Reducer State changes should set returnConfig on "SetReturnArticlesConfig" 1`] = `
Object {
  "articlesCount": 2,
  "currentPage": 1,
  "errorLoadingArticles": false,
  "errorLoadingTags": false,
  "favoritingArticle": null,
  "ids": Array [
    "slug",
    "1",
  ],
  "items": Object {
    "1": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
    "slug": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
  },
  "loadingArticles": false,
  "loadingTags": false,
  "returnConfig": Object {
    "filters": Object {
      "author": "",
      "favorited": "",
      "limit": 10,
      "offset": 0,
      "pageIndex": 0,
      "tag": "",
    },
    "type": "all",
  },
  "tags": Array [],
}
`;

exports[`Article Reducer State changes should set tags on "LoadTagsLoaded" 1`] = `
Object {
  "articlesCount": 2,
  "currentPage": 1,
  "errorLoadingArticles": false,
  "errorLoadingTags": false,
  "favoritingArticle": null,
  "ids": Array [
    "slug",
    "1",
  ],
  "items": Object {
    "1": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
    "slug": Object {
      "author": Object {
        "bio": "bio",
        "following": false,
        "image": "image",
        "isFollowLoading": false,
        "username": "username",
      },
      "body": "body",
      "createdAt": "1544505543896",
      "description": "desc",
      "favorited": false,
      "favoritesCount": 0,
      "favoriting": false,
      "slug": "slug",
      "tagList": Array [
        "list",
      ],
      "title": "title",
      "updatedAt": "1544505523896",
    },
  },
  "loadingArticles": false,
  "loadingTags": false,
  "returnConfig": null,
  "tags": Array [
    "one",
    "two",
  ],
}
`;
